src/utils/string/StringUtils.js

```diff
@@ -240,6 +242,117 @@ const getAnonymousChatName = async (members) => {
   });
 };
 
+const helperGetMemberName = (member) => {
+  const isAnonymous = Boolean(member?.anon_user_info_emoji_name);
+
+  return isAnonymous
+    ? `Anonymous ${member?.anon_user_info_emoji_name}`
+    : member?.user?.username ?? member?.user?.name;
+};
+
+const helperIsSelf = (userId, selfSignUserId, selfAnonUserId) => {
+  return userId === selfSignUserId || userId === selfAnonUserId;
+};
+
+const helperGetWhichMembers = (channel, selfSignUserId, selfAnonUserId) => {
+  const members = [];
+  if (channel?.better_channel_member) {
+    if (isArray(channel?.better_channel_member)) {
+      _.forEach(channel?.better_channel_member, (member) => {
+        const isNotSelf = !helperIsSelf(member?.user_id, selfSignUserId, selfAnonUserId);
+        if (isNotSelf) {
+          members.push(member);
+        }
+      });
+
+      return members;
+    }
+
+    _.forIn(channel?.better_channel_member, (member, key) => {
+      const isNotSelf = !helperIsSelf(key, selfSignUserId, selfAnonUserId);
+      if (isNotSelf) {
+        members.push(member);
+      }
+    });
+
+    return members;
+  }
+
+  channel?.members?.forEach((item) => {
+    const isNotSelf = !helperIsSelf(item?.user_id, selfSignUserId, selfAnonUserId);
+    if (isNotSelf) {
+      members.push(item);
+    }
+  });
+
+  return members;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block contains repetitive logic for pushing members into an array if they are not self. This could be abstracted into a separate function to avoid repetition and improve readability.
```diff
- const helperIsSelf = (userId, selfSignUserId, selfAnonUserId) => {
-   return userId === selfSignUserId || userId === selfAnonUserId;
- };
- 
- const helperGetWhichMembers = (channel, selfSignUserId, selfAnonUserId) => {
-   const members = [];
-   if (channel?.better_channel_member) {
-     if (isArray(channel?.better_channel_member)) {
-       _.forEach(channel?.better_channel_member, (member) => {
-         const isNotSelf = !helperIsSelf(member?.user_id, selfSignUserId, selfAnonUserId);
-         if (isNotSelf) {
-           members.push(member);
-         }
-       });
- 
-       return members;
-     }
- 
-     _.forIn(channel?.better_channel_member, (member, key) => {
-       const isNotSelf = !helperIsSelf(key, selfSignUserId, selfAnonUserId);
-       if (isNotSelf) {
-         members.push(member);
-       }
-     });
- 
-     return members;
-   }
- 
-   channel?.members?.forEach((item) => {
-     const isNotSelf = !helperIsSelf(item?.user_id, selfSignUserId, selfAnonUserId);
-     if (isNotSelf) {
-       members.push(item);
-     }
-   });
- 
-   return members;
- };

+ const helperIsSelf = (userId, selfSignUserId, selfAnonUserId) => {
+   return userId === selfSignUserId || userId === selfAnonUserId;
+ };
+ 
+ const pushIfNotSelf = (member, selfSignUserId, selfAnonUserId, members) => {
+   const isNotSelf = !helperIsSelf(member?.user_id, selfSignUserId, selfAnonUserId);
+   if (isNotSelf) {
+     members.push(member);
+   }
+ };
+ 
+ const helperGetWhichMembers = (channel, selfSignUserId, selfAnonUserId) => {
+   const members = [];
+   if (channel?.better_channel_member) {
+     if (isArray(channel?.better_channel_member)) {
+       _.forEach(channel?.better_channel_member, (member) => {
+         pushIfNotSelf(member, selfSignUserId, selfAnonUserId, members);
+       });
+     } else {
+       _.forIn(channel?.better_channel_member, (member) => {
+         pushIfNotSelf(member, selfSignUserId, selfAnonUserId, members);
+       });
+     }
+   } else if (channel?.members) {
+     channel.members.forEach((item) => {
+       pushIfNotSelf(item, selfSignUserId, selfAnonUserId, members);
+     });
+   }
+ 
+   return members;
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->